tests/PHPStan/Analyser/data/list-type.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace ListType;
+
+use function PHPStan\Analyser\assertType;
+
+class Foo
+{
+	public function withoutGenerics(): void
+	{
+		/** @var list $list */
```

Comment:
You should at least once typehint the type as method parameter and assert it right at the beginning of the body. If you assert the variable only after some changes (array appending), you have no guarantees about the state right after `$list = [];`.

Reply:
Ok, I added some test cases with direct assertions. It should be ok